File: api/src/main/java/com/grash/model/Vehicle.java
package com.grash.model;

import com.grash.model.abstracts.CompanyAudit;
import com.grash.model.enums.fleet.FuelType;
import com.grash.model.enums.fleet.VehicleStatus;
import lombok.Data;
import lombok.NoArgsConstructor;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

@Entity
@Data
@NoArgsConstructor
public class Vehicle extends CompanyAudit {

    @NotNull
    private String name;

    private String assetNumber;

    @Column(unique = false)
    private String vin;

    private String make;
    private String model;
    private Integer year;
    private String trim;
    private String engineType;
    private String transmission;
    private String driveType;
    private String bodyClass;

    private String licensePlate;
    private String color;
    private Integer mileage;

    @Enumerated(EnumType.STRING)
    private FuelType fuelType = FuelType.GASOLINE;

    @Enumerated(EnumType.STRING)
    private VehicleStatus status = VehicleStatus.ACTIVE;

    private String notes;

    @ManyToOne(fetch = FetchType.LAZY)
    private OwnUser assignedDriver;

    @ManyToOne(fetch = FetchType.LAZY)
    private File image;
}

File: mobile/models/vehicle.ts
import { AssetMiniDTO } from './asset';

export type VehicleStatus =
  | 'ACTIVE'
  | 'IN_MAINTENANCE'
  | 'OUT_OF_SERVICE'
  | 'RETIRED';

export default interface Vehicle {
  id: number;
  name: string;
  assetNumber?: string;
  asset?: AssetMiniDTO | null;
  vin?: string;
  make?: string;
  model?: string;
  year?: number;
  licensePlate?: string;
  mileage?: number;
  fuelType?: string;
  status: VehicleStatus;
  notes?: string;
  usageUnit?: 'MILES' | 'HOURS';
}
